project2/kernel.c

```diff
@@ -302,17 +423,21 @@ static void Next_Kernel_Request() {
         Kernel_Request_GetArg,
         Kernel_Request_GetPid,
         Kernel_Request_GetNow,
+        Kernel_Msg_Send,
+        Kernel_Msg_Recv,
+        Kernel_Msg_Rply,
+        Kernel_Msg_ASend,
         Kernel_Request_Terminate
     };
 
     Dispatch();  /* select a new task to run */
 
-    while(1) {
+    for(;;) {
```

Comment:
I like the `for(;;)` style better, thanks for changing it

Reply:
Apparently, it is faster too because no comparison needs to be made